tests/unit/test_space.py

```diff
@@ -114,6 +123,31 @@ def test_discrete_search_space_contains_all_its_points(point: tf.Tensor) -> None
 )
 def test_discrete_search_space_does_not_contain_other_points(point: tf.Tensor) -> None:
     assert point not in DiscreteSearchSpace(_points_in_2D_search_space())
+    assert not DiscreteSearchSpace(_points_in_2D_search_space()).contains(point)
+
+
+def test_discrete_search_space_contains_some_points_but_not_others() -> None:
+    points = tf.constant([[-1.0, -0.4], [-1.0, 0.4], [-1.0, 0.5]])
```

Comment:
Might be worth a note specifying that this test will break if the return of _points_in_2D_search_space changes?